### PR TITLE
Clean up drop table analyzers

### PR DESCRIPTION
**Pull request by mfussenegger**

### CODE REVIEW

**Comment by mfussenegger** [comment]
@seut please review.
the tests that I deleted with DropTableAnalyzedStatementTest.java were duplicated in DropTableAnalyzerTest

File: sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java [hunk 1]
```diff
@@ -21,37 +21,39 @@
...
-            Operation.blockedRaiseException(statement.tableInfo, Operation.DROP);
+    public DropTableAnalyzedStatement analyze(DropTable node, @Nullable String defaultSchema) {
+        TableIdent tableIdent = TableIdent.of(node.table(), defaultSchema);
+        DocTableInfo tableInfo = null;
+        boolean isNoop = false;
+        try {
+            tableInfo = schemas.getDropableTable(tableIdent);
+        } catch (ResourceUnknownException e) {
+            if (node.ignoreNonExistentTable()) {
+                isNoop = true;
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java]: isNoop was taken from the statement before, is the new logic valid?

File: sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java [hunk 2]
```diff
@@ -21,37 +21,39 @@
...
-
-        return statement;
-    }
-
-    public AnalyzedStatement analyze(Node node, Analysis analysis) {
-        return process(node, analysis);
+        if (!isNoop) {
+            Operation.blockedRaiseException(tableInfo, Operation.DROP);
+        }
+        return new DropTableAnalyzedStatement(tableInfo, isNoop, node.ignoreNonExistentTable());
```
Comment (seut) [inline sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java]: ignoreNonExistentTable turns into a dropIfExists var which looks a bit odd. would be great to clean this up as well ;)

File: sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java (hunk 1 above)
Comment (mfussenegger) [inline sql/src/main/java/io/crate/analyze/DropTableAnalyzer.java]: yep it's the same logic

**Pull request merged**